code/game/machinery/ecto_sniffer.dm

```diff
@@ -13,10 +13,16 @@
 	var/sensor_enabled = TRUE
 	///List of ckeys containing players who have recently activated the device, players on this list are prohibited from activating the device untill their residue decays.
 	var/list/ectoplasmic_residues = list()
+	var/obj/item/radio/radio
```

Comment:
Delete the radio in `Destroy()`.
Where it says `QDEL_NULL(wires)` add `QDEL_NULL(radio)` above the `. = ..()`

Reply:
Thanks! Fixed that.